libdnf/conf/OptionStringList.cpp

```diff
@@ -64,18 +64,20 @@ OptionStringList::ValueType OptionStringList::fromString(const std::string & val
 {
     std::vector<std::string> tmp;
     auto start = value.find_first_not_of(" ");
-    if (start == value.npos)
-        return tmp;
-    while (start < value.length()) {
+    while (start != value.npos && start < value.length()) {
         auto end = value.find_first_of(" ,\n", start);
         if (end == value.npos) {
             tmp.push_back(value.substr(start));
             break;
         }
         tmp.push_back(value.substr(start, end - start));
         start = value.find_first_not_of(" ", end + 1);
-        if (start == value.npos && value[end] == ',')
-            tmp.push_back("");
+        if (start != value.npos && value[start] == ',' && value[end] == ' ') {
+            end = start;
+            start = value.find_first_not_of(" ", start + 1);
+        }
+        if (start != value.npos && value[start] == '\n' && (value[end] == ' ' || value[end] == ','))
+            start = value.find_first_not_of(" ", start + 1);
```

Comment:
I'll leave the review to @m-blaha, just a general question, did you consider writing unit tests for the config parser? I think it would be quite easy and very useful here.

Reply:
Yes, to have unit tests here would be really nice.